pytest_fixtures/core/contenthosts.py

```diff
@@ -220,3 +220,9 @@ def sat_ready_rhel(request):
     deploy_args['target_memory'] = '20GiB'
     with Broker(**deploy_args, host_class=ContentHost) as host:
         yield host
+
+
+@pytest.fixture(scope="module")
+def rhel_upgrade_content_host():
+    """A module-level fixture that provides a UBI_8 content host for upgrade scenario testing"""
+    return Broker(container_host='ubi8:latest', host_class=ContentHost).checkout()
```

Comment:
When is this checked in?

Reply:
This is currently handled using CI to remove these Containers as these are needed after the upgrade. I was also thinking of adding the pytest hook to remove these parts of the post upgrade.

Reply:
is CI performing a sync of the Container provider as well? or is this handled by a general checkin of the active inventory after each job?

Reply:
This is handled by a general check-in of the active inventory after each job. (brokerUtils.checkin()). Which copies the inventory file with all container list and do `broker checkin --all`. I still need to implement this layer in the robottelo after the post-upgrade to clean entities (future improvement).